### DISCUSSION TITLE
Is it mandatory that the webserver "airflow" user  GID is "root"

### DISCUSSION THREAD

**Discussion by BELAGGOUN** [airflow-helm/charts#702] [Questions & Answers]
Hello,
I m  deploying Airflow in the Openshift cluster where we have security.context restriction, which rejects deployment if Pods runAsUser and runAsGroupe, and fsGroup are not within a specific range. 
I noticed that the webserver "airflow" user is created with the root privileges (Uid =50000, GID= 0).
Is that a requirement, to have airflow part of the root group? if yes that will be a security issue.( our security team accepted to add an exception for the UID =500000 but not for the GID=0)
I have tried to change the security.context values to the allowed values but I noticed that a "default" user was created and if I move dags to the dag folder, the will have as user "default" not "airflow"
is there any alternative solution where I can deploy the helm chart with the airflow user without root privileges?
Thank you.

**Comment by thesuperzapper** [comment]
@BELAGGOUN the gid is set with the `airflow.image.gid` value, and the reason it defaults to `0`, is because this is what the [airflow image defaults to](https://airflow.apache.org/docs/docker-stack/entrypoint.html#allowing-arbitrary-user-to-run-the-container).

You can change the `airflow.image.gid` value to whatever you want, but its important for your security team to understand that having the "root" group actually provides no permissions (other than access to files with group-level permissions that are owned by gid=0), and it's hard to think of a situation where this is a security risk.

If you can think of an issue with using gid=0, please share it here, and with the airflow community as a whole, so it can be resolved.
